### PR TITLE
One line cr bot

### PR DESCRIPTION
**Pull request by nmanthey**

### CODE REVIEW

File: .github/workflows/one-line-cr-bot.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# branch of a package.
+#
+# The workflow has locations labeled '[ACTION REQUIRED]' where adaptation for
+# your build might be required, as well as where to compare the findings to.
+#
+# To learn more about the available options, check the CLI parameters of the
+# script 'one-line-cr-bot.sh' in https://github.com/awslabs/one-line-scan.git
+name: One Line CR Bot
+
+on:
```
Comment (wipawel) [inline .github/workflows/one-line-cr-bot.yml:13]: I would also add `push` event (as a final state check)

File: .github/workflows/one-line-cr-bot.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      with:
+        fetch-depth: 0
+
+    # one-line-cr-bot.sh will get infer and cppcheck, if not available
+    - name: Install CppCheck Package
+      env:
+        # This is needed in addition to -yq to prevent apt-get from asking for user input
+        DEBIAN_FRONTEND: noninteractive
+      # [ACTION REQUIRED] Add your build dependencies here, drop cppcheck to get latest cppcheck
+      run: |
```
Comment (wipawel) [inline .github/workflows/one-line-cr-bot.yml:39]: Q: what is the `|` for?

File: .github/workflows/one-line-cr-bot.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    # Get the comare remote
+    - name: Setup Compare Remote
+      # [ACTION REQUIRED] Add the https URL of your repository
+      run: git remote add compare https://github.com/awslabs/ktf.git
+    - name: Fetch Compare Remote
+      run: git fetch compare
+
+    # Get one-line-scan, the tool we will use for analysis
+    - name: Get OneLineScan
+      run:  git clone -b one-line-cr-bot https://github.com/nmanthey/one-line-scan.git ../one-line-scan
```
Comment (wipawel) [inline .github/workflows/one-line-cr-bot.yml:51]: Suggestion: you can actually construct a github workflow integrating all these and KTF would just use it (that would make this step and the Setup Compare Remote step much easier)

Comment (wipawel) [reply .github/workflows/one-line-cr-bot.yml:51]: Might be helpful: https://medium.com/sysf/how-to-set-up-github-workflows-and-create-github-actions-using-docker-3a5ba7ec0988

File: .github/workflows/one-line-cr-bot.yml (hunk 2 above)
Comment (nmanthey) [reply .github/workflows/one-line-cr-bot.yml:39]: That's a left-over for multi-line statements. I can likely drop that, or, on the other hand, make it simpler for others to add their lines below.

File: .github/workflows/one-line-cr-bot.yml (hunk 3 above)
Comment (nmanthey) [reply .github/workflows/one-line-cr-bot.yml:51]: Agreed, good idea. I would have to learn more about the interface for that, before being able to properly split relevant things.

Before doing that, I'd try to support a few projects in a manual way.

File: .github/workflows/one-line-cr-bot.yml (hunk 1 above)
Comment (nmanthey) [reply .github/workflows/one-line-cr-bot.yml:13]: for 'push', I am not sure how to make the comparison work nicely. If it's triggered before the push, the upstream compare branch should be good enough.

**Pull request closed without merging**